File: server/talk-talk/src/main/java/com/grabieckacper/talktalk/request/CreateUserRequest.java
package com.grabieckacper.talktalk.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

public record CreateUserRequest(
        @Email
        @NotNull
        String email,

        @NotBlank
        @Size(min = 8)
        String password,

        @NotBlank
        @JsonProperty("first_name")
        String firstName,

        @NotBlank
        @JsonProperty("last_name")
        String lastName
) {}

File: android/talk-talk/app/src/main/java/com/grabieckacper/talktalk/request/CreateUserRequest.kt
package com.grabieckacper.talktalk.request

import com.google.gson.annotations.SerializedName

data class CreateUserRequest(
    val email: String,
    val password: String,
    @SerializedName("first_name")
    val firstName: String,
    @SerializedName("last_name")
    val lastName: String
)
